### PR TITLE
Improve NoSuchFileException usage and robustness

### PR DESCRIPTION
**Pull request by adamnovak**
In #2589 there's evidence that NoSuchFileException might be getting constructed with too many arguments somewhere, masking a real error. This fixes that, so the real error might become apparent.

I couldn't find any code that actually seemed to be doing that, but this makes the exception handle extra arguments being dumped into it.

### CODE REVIEW

File: src/toil/jobStores/abstractJobStore.py [hunk 1]
```diff
@@ -80,15 +80,24 @@ def __init__(self, jobStoreFileID):
...
+        # most three arguments error reported in
+        # https://github.com/DataBiosphere/toil/issues/2589#issuecomment-481912211
         if customName is None:
             message = "File '%s' does not exist." % jobStoreFileID
         else:
             message = "File '%s' (%s) does not exist." % (customName, jobStoreFileID)
+        
+        if extra:
+            # Append extra data.
+            message += " " + " ".join((str(x) for x in extra))
```
Comment (DailyDreaming) [inline src/toil/jobStores/abstractJobStore.py]: Would prepending a string here explaining what this is trying to solve be good to have in front of the mysterious args?

Comment (adamnovak) [reply src/toil/jobStores/abstractJobStore.py]: OK I put a generic "Extra info:" message. Do you think that works? Or do we want to complain that the extra info shouldn't be provided?

Comment (DailyDreaming) [reply src/toil/jobStores/abstractJobStore.py]: I think that ought to do the trick.  At least you know it's intentional and not something else leaking over.

**Comment by mr-c** [comment]
Huh, why isn't  GitLab testing this?

**Comment by adamnovak** [comment]
Gitlab has decided it doesn't like this branch. Pussing the same commit to another branch kicked off the tests.

**Pull request merged**